colossalai/tensor/_ops/layernorm.py

```diff
@@ -0,0 +1,37 @@
+from numpy import isin, kaiser
+import torch
+from colossalai.tensor.op_wrapper import colo_op_impl
+from colossalai.tensor import ColoTensor
+
+
+@colo_op_impl(torch.nn.functional.layer_norm)
+def colo_layernorm(types, args=(), kwargs=None, pg=None):
+    arg_num = len(args)
+    if arg_num > 0:
+        input = args[0]
+    if arg_num > 1:
+        normalized_shape = args[1]
+    if arg_num > 2:
+        weight = args[3]
+    if arg_num > 3:
+        bias = args[4]
+    if arg_num > 4:
+        eps = args[5]
+
+    if 'input' in kwargs:
+        input = kwargs['input']
+    if 'weight' in kwargs:
+        weight = kwargs['weight']
+    if 'bias' in kwargs:
+        bias = kwargs['bias']
+    if 'eps' in kwargs:
+        eps = kwargs['eps']
+
+    if isinstance(input, ColoTensor):
+        input = input.torch_tensor()
+    if isinstance(weight, ColoTensor):
+        weight = weight.torch_tensor()
+    if isinstance(bias, ColoTensor):
+        bias = bias.torch_tensor()
+
+    return ColoTensor.init_from_torch_tensor(torch.nn.functional.layer_norm(input, normalized_shape, weight, bias, eps))
```

Comment:
It's 1D-specific layernorm.

Reply:
You mean TP 1D row splitting?

Reply:
You suggest I assign a Spec to the weight/bias tensor of LN? and add a if for weight.spec to make sure the logic is only work for 1D row?